homeassistant/components/reolink/__init__.py

```diff
@@ -106,9 +106,10 @@ async def async_check_firmware_update():
     )
     # Fetch initial data so we have data when entities subscribe
     try:
+        # If camera WAN blocked, firmware check fails, do not prevent setup
```

Comment:
Can we check this before we activate the firmware coordinator?

Reply:
Not really, the only way to check if WAN is blocked is just try out a command that needs WAN such as the check_new_firmware command and the API will respond with `"rspCode" : -17, "detail" : "rcv failed",`.
However I have seen this `rcv failed` error before, also when you have a bad wifi connection to the camera or other issues.

Reply:
Ok. It's just a bit annoying that we need to log an error for a case where users have deliberately blocked WAN access.

Reply:
@MartinHjelmare I could remove the coordinator initial fetch from __init__.py and add a initial fetch to async_added_to_hass in the update entity. That way when the user disables the update entity because they blocked WAN, no more errors are logged.
What do you think of that?

Reply:
The coordinator is meant to fetch data for more than a single entity. Fetching data from the entity isn't in line with the coordinator architecture. If there's a single entity for the data source we don't really need a coordinator.